File: frontend/demo/component/checkbox/checkbox-disabled.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/checkbox';
import '@vaadin/checkbox-group';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('checkbox-disabled')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-checkbox-group label="Departments" theme="vertical" disabled>
        <vaadin-checkbox value="engineering" label="Engineering"></vaadin-checkbox>
        <vaadin-checkbox value="human-resources" label="Human Resources"></vaadin-checkbox>
        <vaadin-checkbox value="marketing" label="Marketing"></vaadin-checkbox>
        <vaadin-checkbox value="operations" label="Operations"></vaadin-checkbox>
        <vaadin-checkbox value="sales" label="Sales"></vaadin-checkbox>
      </vaadin-checkbox-group>
      <!-- end::snippet[] -->
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/checkbox/CheckboxDisabled.java
package com.vaadin.demo.component.checkbox;

import com.vaadin.flow.component.checkbox.CheckboxGroup;
import com.vaadin.flow.component.checkbox.CheckboxGroupVariant;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

@Route("checkbox-disabled")
public class CheckboxDisabled extends Div {

    public CheckboxDisabled() {
        // tag::snippet[]
        CheckboxGroup<String> disabledCheckGroup = new CheckboxGroup<>();
        disabledCheckGroup.setLabel("Departments");
        disabledCheckGroup.setItems("Engineering", "Human Resources", "Marketing", "Operations", "Sales");
        disabledCheckGroup.addThemeVariants(CheckboxGroupVariant.LUMO_VERTICAL);
        disabledCheckGroup.setEnabled(false);
        add(disabledCheckGroup);
        // end::snippet[]
    }

    public static class Exporter extends DemoExporter<CheckboxDisabled> { // hidden-source-line
    } // hidden-source-line
}
